### PR TITLE
Update prod-util recipe

### PR DESCRIPTION
**Pull request by AlexanderRichert-NOAA**
This PR adds the latest versions of prod-util, removes an old one, and updates the dependency from w3nco to w3emc.

### CODE REVIEW

File: var/spack/repos/builtin/packages/prod-util/package.py [hunk 1]
```diff
@@ -14,9 +14,16 @@ class ProdUtil(CMakePackage):
 
     homepage = "https://github.com/NOAA-EMC/NCEPLIBS-prod_util"
     url = "https://github.com/NOAA-EMC/NCEPLIBS-prod_util/archive/refs/tags/v1.2.2.tar.gz"
+    git = "https://github.com/NOAA-EMC/NCEPLIBS-prod_util"
 
     maintainers("AlexanderRichert-NOAA", "Hang-Lei-NOAA", "edwardhartnett")
 
-    version("1.2.2", sha256="c51b903ea5a046cb9b545b5c04fd28647c58b4ab6182e61710f0287846350ef8")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/prod-util/package.py:20]: The general guidance (https://spack.readthedocs.io/en/latest/packaging_guide.html#deprecating-old-versions) is to deprecate older versions before removing them thought there are exceptions.  Let me know if you want to deprecate or really remove this.

**Review by tldahlgren — DISMISSED** [review]
Confirmed the branch and new version sha256s.  Just a quick question about deprecating versus removing the older version.

File: var/spack/repos/builtin/packages/prod-util/package.py (hunk 1 above)
Comment (AlexanderRichert-NOAA) [reply var/spack/repos/builtin/packages/prod-util/package.py:20]: Good point, I'm deprecating but keeping the older version and updating deps accordingly.

**Comment by AlexanderRichert-NOAA** [comment]
@spackbot fix style

**Review by tldahlgren — APPROVED** [review]

**Review by edwardhartnett — APPROVED** [review]

**Comment by tldahlgren** [comment]
Just waiting on CI to finish.

**Pull request merged**